File: CleverFerretV2/core/database/src/main/java/com/cleverferret/v2/core/database/schema/MediaItemV1.java
package com.cleverferret.v2.core.database.schema;

import com.cleverferret.v2.core.common.VersionedContract;
import com.cleverferret.v2.core.data.contracts.MediaTypeV1;

/**
 * Canonical media identity row shared by all format-specific variants.
 */
public record MediaItemV1(
        String mediaItemId,
        String libraryId,
        MediaTypeV1 mediaType,
        String title,
        String normalizedTitle,
        String primaryCreator,
        boolean favorite,
        long createdAtEpochMs,
        long updatedAtEpochMs,
        long lastAccessedAtEpochMs
) implements VersionedContract {
    @Override
    public String contractVersion() {
        return "v1";
    }
}

File: CleverFerretV2/core/database/src/main/java/com/cleverferret/v2/core/database/schema/MediaItemV1.kt
package com.cleverferret.v2.core.database.schema

import com.cleverferret.v2.core.common.VersionedContract
import com.cleverferret.v2.core.data.contracts.MediaTypeV1

data class MediaItemV1(
    val mediaItemId: String,
    val libraryId: String,
    val mediaType: MediaTypeV1,
    val title: String,
    val normalizedTitle: String,
    val primaryCreator: String,
    val favorite: Boolean,
    val createdAtEpochMs: Long,
    val updatedAtEpochMs: Long,
    val lastAccessedAtEpochMs: Long
) : VersionedContract {
    override fun contractVersion(): String = "v1"
}
